cli/src/cli_util.rs

```diff
@@ -1448,6 +1551,34 @@ impl WorkspaceCommandTransaction<'_> {
     pub fn into_inner(self) -> Transaction {
         self.tx
     }
+
+    /// Moves each branch in `branches` from an old commit it's associated with
+    /// (configured by `get_advanceable_branches`) to the `move_to` commit. If
+    /// the branch is conflicted before the update, it will remain conflicted
+    /// after the update, but the conflict will involve the `move_to` commit
+    /// instead of the old commit.
+    pub fn advance_branches(&mut self, branches: Vec<AdvanceableBranch>, move_to: &CommitId) {
+        for branch in branches {
+            // We are going to remove the old commit and add the new commit. The
+            // removed commit must be listed first in order for the `MergeBuilder`
+            // to recognize that it's being removed.
+            let remove_add = [Some(branch.old_commit_id), Some(move_to.clone())];
+            let new_target = RefTarget::from_merge(
+                MergeBuilder::from_iter(
+                    branch
+                        .old_target
+                        .as_merge()
+                        .iter()
+                        .chain(&remove_add)
+                        .cloned(),
+                )
+                .build()
+                .simplify(),
+            );
+            self.mut_repo()
+                .set_local_branch_target(&branch.name, new_target);
```

Comment:
Perhaps, `.merge_local_branch()` can be used.

```rust
self.mut_repo().merge_local_branch(
    &branch.name,
    &RefTarget::normal(branch.old_commit_id.clone()),
    &RefTarget::normal(move_to.clone()),
);
```

(I thought I added this comment in the previous round, but apparently not.)

Reply:
Thanks, this works perfectly.